csl_demo/BlockfrostProvider.gd

```diff
@@ -88,51 +82,70 @@ func blockfrost_request(request: Request) -> Variant:
 		remove_child(http_request)
 		return {}
 
-	var result = await http_request.request_completed
+	var result : Array = await http_request.request_completed
+	var status_code : int = result[1]
+	var content_bytes : PackedByteArray = result[3]
+	var content := content_bytes.get_string_from_utf8()
 	remove_child(http_request)
 	
 	# TODO: handle error responses properly
-	if result[1] != 200:
-		print("Blockfrost request failed: ", result[3].get_string_from_utf8())
+	if status_code != 200:
+		print("Blockfrost request failed: ", content)
 		return null
 		
-	return JSON.parse_string(result[3].get_string_from_utf8())
+	return JSON.parse_string(content)
 
 func get_protocol_parameters() -> ProtocolParameters:
 	var params_json: Dictionary = await blockfrost_request(ProtocolParametersRequest.new(current_epoch))
-	var params = ProtocolParameters.create(
-		int(params_json["coins_per_utxo_size"]),
-		int(params_json["pool_deposit"]),
-		int(params_json["key_deposit"]),
-		int(params_json["max_val_size"]),
-		int(params_json["max_tx_size"]),
-		int(params_json["min_fee_b"]),
-		int(params_json["min_fee_a"])
+	# Type hints and dictionaries don't interact well...
+	# We have to cast the dictionary values to [Variant] before [int], otherwise
+	# it will complain about the operation not being safe.
+	var params := ProtocolParameters.create(
+		params_json.coins_per_utxo_size as Variant as int,
+		params_json.pool_deposit as Variant as int,
+		params_json.key_deposit as Variant as int,
+		params_json.max_val_size as Variant as int,
+		params_json.max_tx_size as Variant as int,
+		params_json.min_fee_b as Variant as int,
+		params_json.min_fee_a as Variant as int
 	)
 	self.got_protocol_parameters.emit(params)
 	return params
 
 func utxo_assets(utxo: Dictionary) -> Dictionary:
 	var assets: Dictionary = {}
-	utxo.amount.map(
-		func(asset): assets[asset.unit] = BigInt.from_str(asset.quantity)
+	var amount: Array = utxo.amount
+	var _ret := amount.map(
+		func(asset: Dictionary) -> void:
+			var quantity: String = asset.quantity
+			var res : BigInt.ConversionResult = BigInt.from_str(quantity)
+			if res.is_ok():
+				# We have to return a [_BigInt] here because that is what the Rust code
+				# expects. This should be fixed from the Rust side of things.
+				assets[asset.unit] = res.value._b
+			else:
+				push_error("There was an error while reading the assets from a utxo", res.error)
```

Comment:
They don't really need to check this for any `Result`, they can still just access the result with `res.value`. But they are risking a Rust panic by doing so.

In this case it makes sense to check for the error, since we don't know the String being passed.

Reply:
Good idea!